internal/p2p/peermanager.go

```diff
@@ -527,20 +527,20 @@ func (m *PeerManager) DialNext(ctx context.Context) (NodeAddress, error) {
 
 // TryDialNext is equivalent to DialNext(), but immediately returns an empty
 // address if no peers or connection slots are available.
-func (m *PeerManager) TryDialNext() (NodeAddress, error) {
+func (m *PeerManager) TryDialNext() NodeAddress {
```

Comment:
Part of me would like to see this return a `bool` alongside the result so we don't have to do a zero-value comparison with an empty struct at the usage.  That said—that type isn't super likely to grow/change a lot.